Added referrer attribution columns to events table

rshbhgrg opened the pull request:
refs https://github.com/TryGhost/Team/issues/1931

- referrer source and medium along with url will need to be be stored in the events table along with rest of attribution data
- stores information on two event tables
  - `members_created_events` for newly created members(both free and paid)
  - `members_subscription_created_events` for new paid subscription

Note: Source and Medium are created by us and are not expected to be too long, so have 191 char limit. Referrer URL is stored as Top-level domain only

rishabhgrg commented on ghost/core/core/server/data/migrations/versions/5.16/2022-09-19-17-44-add-referrer-attribution-columns-to-events-table.js:
```diff
@@ -0,0 +1,39 @@
+        maxlength: 2000,
+        nullable: true
+    })
+);
```
@daniellockyer Does all the columns in single migration work better here or should i break this into two separate migrations for two tables ?

daniellockyer replied:
Although we have no strict rules here, I think it might be better to keep it to 1 migration, 1 table

Therefore, maybe let's split them apart?

rishabhgrg replied:
Yea sounds good, updating

rishabhgrg replied:
Updated now ✅

ErisDS commented on ghost/core/core/server/data/schema/schema.js:
```diff
@@ -497,6 +497,9 @@ module.exports = {
         attribution_url: {type: 'string', maxlength: 2000, nullable: true},
+        attribution_ref_source: {type: 'string', maxlength: 191, nullable: true},
+        attribution_ref_medium: {type: 'string', maxlength: 191, nullable: true},
+        attribution_ref_url: {type: 'string', maxlength: 2000, nullable: true},
```
What's the difference between attribution_url and attribution_ref_url?

rishabhgrg replied:
@ErisDS `attribution_url` is the Ghost site URL from which member signed up or paid, for ex. a specific post or site home page. 
`attribution_ref_url`  captures the referrer's URL which led member to that particular post(or other) Ghost site URL, like Ghost Explore or Twitter.

ErisDS replied:
OK - I think we might want to rename these fields to `referrer_url`, `referrer_medium` and `referrer_source` and keep `attribution` to mean our internal post attribution?

ErisDS replied:
@allouis do you agree?

rishabhgrg replied:
Main reason for keeping the `attribution_` prefix was to have all attribution data(internal and external) grouped together for the event, but given the above confusion rename sounds good to me 👍

allouis replied:
@ErisDS I do 👍

ErisDS replied:
Also whilst we're here, before we have any "fun" the correct spelling is "referrer" with 2 R's... or 4 R's if you're a pedant 😂

rishabhgrg replied:
Haha all done to make sure we have the right "referrer" 😄

rshbhgrg committed:
Added referrer attribution columns to events table

refs TryGhost/Team#1931

- referrer source and medium along with url will need to be be stored in the events table along with rest of attribution data
- stores source information on two tables
  - `members_created_events` for newly created members(both free and paid)
  -  `members_subscription_created_events` for new paid subscription

rishabhgrg commented:
@ErisDS @daniellockyer This is ready for review again after above changes

ErisDS reviewed (approved):
LGTM 👍
